app/controllers/application_controller/timelines.rb

```diff
@@ -127,41 +127,12 @@ def tl_chooser
     end
   end
 
-  def getTLdata
-    if session[:tl_xml_blob_id] != nil
-      blob = BinaryBlob.find(session[:tl_xml_blob_id])
-      render :xml=>blob.binary
+  def timeline_data
+    if session[:tl_xml_blob_id] && BinaryBlob.find_by_id(session[:tl_xml_blob_id])
+      blob = BinaryBlob.find_by_id(session[:tl_xml_blob_id])
```

Comment:
There is no need to do the same query in the `if` expression AND inside the `if` block.  Here is a suggested optimization:
```ruby
def timeline_data
  return if session[:tl_xml_blob_id].nil?
  blob = BinaryBlob.find_by_id(session[:tl_xml_blob_id])
  return if blob.nil?
  render :xml => blob.binary
  blob.destroy
end
```